### PR TITLE
Document adding new Implementations, Versions and Specifications in the tool. fixes #20.

### LINKED ISSUE #20: Document how MP Implementation can be added to Version dropdown
When a vendor updates the implementation, what steps need to be done to have the implementation show up in the drop-down for a specific MP version.

### PR DESCRIPTION
**Pull request by rdebusscher**

### CODE REVIEW

File: how-to.md [hunk 1]
```diff
@@ -1,8 +1,48 @@
 # How to
 
+## Add new MicroProfile Implementation
+
+What are the technical steps to add a new MicroProfile Implementation?
```
Comment (Emily-Jiang) [inline how-to.md]: I suggest not to have this line as it does not add anything more. We have headings already.

File: how-to.md [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+
+* Add the implementation to the enum **org.eclipse.microprofile.starter.addon.microprofile.servers.model.SupportedServer**   
+    The first parameter is the 'code' for the new implementation.  
+    The second parameter is the value for the dropdown in the UI.  
+    The third parameter is the list of supported MicroProfile Versions
```
Comment (Emily-Jiang) [inline how-to.md]: It will help if you add an example here.

File: how-to.md [hunk 3]
```diff
@@ -1,8 +1,48 @@
...
+* Add a 'profile' to the file `src/main/resources/pom)servers.xml` for each Version specified for this new implementation in the previous step.   
+   The id of the profile must have the structure `<code>-<version>` with the code the value you specified in the previous bullet and version the MicroProfile version this _profile_ must be used for.
```
Comment (Emily-Jiang) [inline how-to.md]: Add an example.

File: how-to.md [hunk 4]
```diff
@@ -1,8 +1,48 @@
...
+
+* Add the implementation to the enum **org.eclipse.microprofile.starter.addon.microprofile.servers.model.SupportedServer**   
+    The first parameter is the 'code' for the new implementation.  
+    The second parameter is the value for the dropdown in the UI.  
+    The third parameter is the list of supported MicroProfile Versions
+* Add a 'profile' to the file `src/main/resources/pom)servers.xml` for each Version specified for this new implementation in the previous step.   
+   The id of the profile must have the structure `<code>-<version>` with the code the value you specified in the previous bullet and version the MicroProfile version this _profile_ must be used for.  
+   This profile should generate an executable jar file.
+* Add the generated executable jar file in the method **defineJarFileName()** and the URL of the root page of the index.xhtml page within the method **defineTestURL** of **org.eclipse.microprofile.starter.addon.microprofile.servers.MicroprofileServersAddon**
+* Test out the generated project with this new MicroProfile implementation.
```
Comment (Emily-Jiang) [inline how-to.md]: How to test out?

File: how-to.md [hunk 5]
```diff
@@ -1,8 +1,48 @@
...
+* In case the implementation requires adjusted example files (which is, however, an indication of some problem following the specification) you can create a customized template within the directory `src/main/resources/files/<code>`.  
+   This adjusted template file will be picked up automatically.
+* If needed, the _pom.xml_ generation and which files are created can be customized through the methods **adaptMavenModel()** and **createFiles()** of **org.eclipse.microprofile.starter.addon.microprofile.servers.MicroprofileServersAddon**
+
+## Add a new MicroProfile Version to an Implementation
+
+How is an Implementation added to another MicroProfile Version?
```
Comment (Emily-Jiang) [inline how-to.md]: Can be deleted.

File: how-to.md [hunk 6]
```diff
@@ -1,8 +1,48 @@
...
+
+* Add the version to the list of supported version of the implementation (see enum **org.eclipse.microprofile.starter.addon.microprofile.servers.model.SupportedServer**)   
+* Add a 'profile' to the file `src/main/resources/pom)servers.xml`.  
+   The id of the profile must have the structure `<code>-<version>` with the code the value you specified in the previous bullet and version the MicroProfile version this profile must be used for.
```
Comment (Emily-Jiang) [inline how-to.md]: Please provide an example.

File: how-to.md [hunk 7]
```diff
@@ -1,8 +1,48 @@
...
+   This profile should generate an executable jar file.
+* Test out the generated project with this new version.
+* In case the implementation requires adjusted example files (which is, however, an indication of some problem following the specification) you can create a customized template within the directory `src/main/resources/files/<code>`
+   This adjusted template file will be picked up automatically.
+
+## Adding new Specification
+
+What steps are required to add a new Specification
+
+* Add an new value to the enum **org.eclipse.microprofile.starter.addon.microprofile.servers.model.MicroprofileSpec**
```
Comment (Emily-Jiang) [inline how-to.md]: an new value -> a new value

File: how-to.md [hunk 8]
```diff
@@ -1,8 +1,48 @@
...
+* Add a 'profile' to the file `src/main/resources/pom)servers.xml`.  
+   The id of the profile must have the structure `<code>-<version>` with the code the value you specified in the previous bullet and version the MicroProfile version this profile must be used for.  
+   This profile should generate an executable jar file.
+* Test out the generated project with this new version.
+* In case the implementation requires adjusted example files (which is, however, an indication of some problem following the specification) you can create a customized template within the directory `src/main/resources/files/<code>`
+   This adjusted template file will be picked up automatically.
+
+## Adding new Specification
+
+What steps are required to add a new Specification
```
Comment (Emily-Jiang) [inline how-to.md]: Can be deleted.

File: how-to.md [hunk 9]
```diff
@@ -1,8 +1,48 @@
...
+
+* Add an new value to the enum **org.eclipse.microprofile.starter.addon.microprofile.servers.model.MicroprofileSpec**    
+     The first parameter is the 'code' for the new specification.  
+     The second parameter is the value for the selection of the specification in the UI.  
+     The third parameter is the list of MicroProfile Version the specification is part of.
+* Add an `if-structure` to the method **createFiles()** of **org.eclipse.microprofile.starter.addon.microprofile.servers.MicroprofileServersAddon** so that files are included in the demo application when the specification is selected.
```
Comment (Emily-Jiang) [inline how-to.md]: Please provide an example.

File: how-to.md [hunk 10]
```diff
@@ -1,8 +1,48 @@
...
+* Update the **readme.md.tpl** to include some basic information about this new specification. Make this addition conditional (see other specification on how to do this)
+
 ## Add new template file
 
-Before new template files can be used within the generated project, following steps need to be perfomed:
+Before new template files can be used within the generated project, following steps need to be performed:
```
Comment (Emily-Jiang) [inline how-to.md]: Please explain what the template file is for. When do you need to add the templates?

Comment (Emily-Jiang) [reply how-to.md]: What info should be in the template?

**Comment by rdebusscher** [comment]
@Emily-Jiang Updated the "how to" documentation based on your remarks.

**Pull request merged**